input/cql/PCSBPScreeningFollowUpFHIR.cql

```diff
@@ -250,26 +251,35 @@ define "NonPharmacological Intervention Not Ordered":
     union [ServiceNotRequested: "Dietary Recommendations"]
     union [ServiceNotRequested: "Recommendation to Increase Physical Activity"]
     union [ServiceNotRequested: "Referral or Counseling for Alcohol Consumption"] ) NonPharmIntervention
-    where NonPharmIntervention.reasonRefused in "Patient Declined"
-      and NonPharmIntervention.status = 'completed'
+      where NonPharmIntervention.status = 'completed'
+        and exists(NonPharmIntervention.extension E
+          where E.url = 'http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-doNotPerformReason' 
+          and E.value as Code in "Patient Declined"
+        )
 
 define "Encounter with Medical Reason for Not Obtaining or Patient Declined Blood Pressure Measurement":
   "Qualifying Encounter during Measurement Period" QualifyingEncounter
-    with ( [ObservationNotDone: code ~ "Systolic blood pressure"]
-      union [ObservationNotDone: code ~ "Diastolic blood pressure"] ) NoBPScreen
+    with ( [ObservationCancelled: code ~ "Systolic blood pressure"]
+      union [ObservationCancelled: code ~ "Diastolic blood pressure"] ) NoBPScreen
       such that NoBPScreen.issued during day of QualifyingEncounter.period
-        and ( NoBPScreen.notDoneReason in "Patient Declined"
-            or NoBPScreen.notDoneReason in "Medical Reason"
-        )
         and NoBPScreen.status = 'cancelled'
+        and exists(NoBPScreen.extension E 
+          where E.url = 'http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-notDoneReason'
+          and (E.value as Code in "Patient Declined"
+            or E.value as Code in "Medical Reason"
+          )
+        )
```

Comment:
We may want to ask Bryn if the `NoBPScreen.notDoneReason` is expected to resolve to the extension NoBPScreen.extension[url='http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-notDoneReason']`

I think those *_should be_* equivalent.

Reply:
That's correct, notDoneReason, since that is the sliceName of the extension